### PR TITLE
Smile - lage nva-sentralimport-submodul

### PR DESCRIPTION
**Pull request by gregorgabriel**
- sub-module for nva-sentralimport: create-publication-from-scopus
- generere klasser for Scopus Publication
- lage ramme for Lambda (det gjenstå utvidelsen av CloudFormation template)

### CODE REVIEW

**Review by anetteOlli — APPROVED** [review]

**Review by anetteOlli — APPROVED** [review]

File: buildSrc/src/main/groovy/nva-fetch-doi.java-common-conventions.gradle [hunk 1]
```diff
@@ -64,13 +64,22 @@ checkstyle {
...
+    exclude { '**/generated/*' }
     reports {
         xml.enabled false
         html.enabled true
         html.stylesheet rootProject.resources.text.fromFile('config/checkstyle/checkstyle-simple.xsl')
     }
 }
 
+tasks.withType(Pmd) {
+    exclude { '**/generated/*' }
```
Comment (brinxmat) [inline buildSrc/src/main/groovy/nva-fetch-doi.java-common-conventions.gradle:76]: Not necessary…see comment on subproject build.gradle

File: buildSrc/src/main/groovy/nva-fetch-doi.java-common-conventions.gradle [hunk 2]
```diff
@@ -92,6 +101,13 @@ jacocoTestCoverageVerification {
             }
         }
     }
+    afterEvaluate {
```
Comment (brinxmat) [inline buildSrc/src/main/groovy/nva-fetch-doi.java-common-conventions.gradle:104]: Ditto

File: create-publication-from-scopus/build.gradle [hunk 3]
```diff
@@ -0,0 +1,48 @@
+def problemVersion = '0.24.0'
+
+project.ext {
```
Comment (brinxmat) [inline create-publication-from-scopus/build.gradle:3]: Don't do this, rather add the generated code to the source set with e.g.

``` 
    project.ext {
        jaxbTargetDir = file("build/generated/sources/xjc")
    }
    sourceSets.main.java.srcDirs += jaxbTargetDir
```

File: create-publication-from-scopus/build.gradle [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+task xsd2java() {
+    doLast {
+        jaxbTargetDir.mkdirs()
+        ant.taskdef(name: 'xjc', classname: 'com.sun.tools.xjc.XJC2Task', classpath: configurations.jaxb.asPath)
+        ant.jaxbTargetDir = jaxbTargetDir
+        ant.xjc(
+                destdir: '${jaxbTargetDir}',
+                package: 'no.scopus.generated',
+                schema: 'src/main/resources/schema.scopus.xocs-ani515/xocs-ani515.xsd',
```
Comment (brinxmat) [inline create-publication-from-scopus/build.gradle:40]: See PR #205

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/PublicationConverterTest.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package no.sikt.nva.scopus;
+
+import org.junit.jupiter.api.Test;
+
+
+class PublicationConverterTest {
```
Comment (brinxmat) [inline create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/PublicationConverterTest.java:6]: Test is unnecessary as it is tested by the handler

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/PublicationPersistenceServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,12 @@
+package no.sikt.nva.scopus;
+
+import org.junit.jupiter.api.Test;
+
+
+class PublicationPersistenceServiceTest {
+
+    @Test
+    public void testNothing() {
```
Comment (brinxmat) [inline create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/PublicationPersistenceServiceTest.java:9]: Should likely be tested by handler

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusS3ClientTest.java [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class ScopusS3ClientTest {
```
Comment (brinxmat) [inline create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusS3ClientTest.java:23]: Tested by handler?

File: gradle/libs.versions.toml [hunk 8]
```diff
@@ -3,7 +3,7 @@ nva = { strictly = "1.16.30" }
 testUtils = { strictly = "0.1.24" }
 jacksonVersion = { strictly = "2.13.0" }
 jakartaVersion = { strictly = "2.1.6" }
-awsSdkVersion = { strictly = "1.12.129" }
+awsSdkVersion = { strictly = "1.12.134" }
```
Comment (brinxmat) [inline gradle/libs.versions.toml:6]: Is there a reason to not use sdk2?

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/PublicationConverterTest.java (hunk 5 above)
Comment (gregorgabriel) [reply create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/PublicationConverterTest.java:6]: It is just a placeholder since the functionality will be implemented later. And we had to give JaCoco a bone to chew on.

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/PublicationPersistenceServiceTest.java (hunk 6 above)
Comment (gregorgabriel) [reply create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/PublicationPersistenceServiceTest.java:9]: It is just a placeholder since the functionality will be implemented later. And we had to give JaCoco a bone to chew on.

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusS3ClientTest.java (hunk 7 above)
Comment (gregorgabriel) [reply create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusS3ClientTest.java:23]: I prefer to test classes/objects separately.  Now I know my class is doing what I expect it to do. And further up in the code (Handler) I can mock the whole thing.

File: gradle/libs.versions.toml (hunk 8 above)
Comment (gregorgabriel) [reply gradle/libs.versions.toml:6]: No. You tell me. We just use what is already in use in nva-fetch-doi

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+    public void testResponseWithEmptyRequest()
+            throws Exception {
+        ScopusHandler handler = createHandler(environment);
+        ByteArrayOutputStream output = new ByteArrayOutputStream();
+        handler.handleRequest(handlerInputStream(), output, context);
+        GatewayResponse<Summary> gatewayResponse = parseSuccessResponse(output.toString());
+        assertEquals(SC_BAD_REQUEST, gatewayResponse.getStatusCode());
+        assertThat(gatewayResponse.getHeaders(), hasKey(CONTENT_TYPE));
+        assertThat(gatewayResponse.getHeaders(), hasKey(HttpHeaders.ACCESS_CONTROL_ALLOW_ORIGIN));
+        assertNull(handler.processInput(null, null, null));
```
Comment (axthosarouris) [inline create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java:62]: Not sure what this line is supposed to test.

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+        assertNull(handler.getSuccessStatusCode(null, null));
```
Comment (axthosarouris) [inline create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java:63]: Not sure what this line is supposed to test.

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+import static org.apache.http.HttpStatus.SC_BAD_REQUEST;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasKey;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class ScopusHandlerTest {
```
Comment (axthosarouris) [inline create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java:35]: I think all the tests should start with testing handleRequest and nothing else. For anything else it is very difficult to follow the scenario. Example ```handler.processInput(null, null, null)``` would never occur.

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+        GatewayResponse<Summary> gatewayResponse = parseSuccessResponse(output.toString());
+        assertEquals(SC_BAD_REQUEST, gatewayResponse.getStatusCode());
+        assertThat(gatewayResponse.getHeaders(), hasKey(CONTENT_TYPE));
+        assertThat(gatewayResponse.getHeaders(), hasKey(HttpHeaders.ACCESS_CONTROL_ALLOW_ORIGIN));
+        assertNull(handler.processInput(null, null, null));
+        assertNull(handler.getSuccessStatusCode(null, null));
+    }
+
+    private GatewayResponse<Summary> parseSuccessResponse(String output) throws JsonProcessingException {
+        return parseGatewayResponse(output, Summary.class);
```
Comment (axthosarouris) [inline create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java:67]: GatewayResponse has static methods for parsing the output 
```
GatewayResponse<Summary> gw = GatewayResponse.fromOutputStream(output);
        Summary = gw.getBodyObject(Summary.class)
```

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+
+
+    private <T> GatewayResponse<T> parseGatewayResponse(String output, Class<T> responseObjectClass)
+            throws JsonProcessingException {
+        JavaType typeRef = restServiceObjectMapper.getTypeFactory()
+                .constructParametricType(GatewayResponse.class, responseObjectClass);
+        return restServiceObjectMapper.readValue(output, typeRef);
+    }
+
+    private InputStream handlerInputStream() {
```
Comment (axthosarouris) [inline create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java:78]: we have the HandlerRequestBuilder for generating input to the handlers

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusS3ClientTest.java [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    @Test
+    public void testConstructorsForCodeCoverage() {
+        ScopusS3Client client = new ScopusS3Client(environment);
+    }
+
+    @Test
+    public void testGetFile() {
+        S3Object s3object = mock(S3Object.class);
```
Comment (axthosarouris) [inline create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusS3ClientTest.java:49]: We have a FakeS3Client that can be used to test the functionality of communicating with S3

File: gradle/libs.versions.toml (hunk 8 above)
Comment (axthosarouris) [reply gradle/libs.versions.toml:6]: We should definetely use sdk2

**Review by axthosarouris — CHANGES_REQUESTED** [review]

File: create-publication-from-scopus/build.gradle (hunk 3 above)
Comment (gregorgabriel) [reply create-publication-from-scopus/build.gradle:3]: Hvorfor det? Vil det virker da å bruke klassene i unmarshalling?

File: create-publication-from-scopus/build.gradle (hunk 4 above)
Comment (gregorgabriel) [reply create-publication-from-scopus/build.gradle:40]: Not sure going to the remote files will improve things. 
We may get updated files that may change behavior, that must be taken care of later. So it may break Test. Than we have to fix the tests. But maybe tahts what we should do anyway...

File: buildSrc/src/main/groovy/nva-fetch-doi.java-common-conventions.gradle (hunk 1 above)
Comment (gregorgabriel) [reply buildSrc/src/main/groovy/nva-fetch-doi.java-common-conventions.gradle:76]: I don't get why think this is unnecessary. Somewhere we have to tell PMD, Checkstyle and JaCoco to ignore those.
I think its best done at the root level.

File: create-publication-from-scopus/README.md [hunk 15]
```diff
@@ -0,0 +1,8 @@
+# CREATE PUBLICATION FROM SCOPUS
+
+## Building:
+NB. It might be needed to run gradle xsd2java separately in order to get the no.scopus.generated package.
```
[the remark this answers is not in the document]
Comment (gregorgabriel) [reply create-publication-from-scopus/README.md]: Yes. We did manage to build it from root gradle level (by now).

File: create-publication-from-scopus/build.gradle (hunk 3 above)
Comment (axthosarouris) [reply create-publication-from-scopus/build.gradle:3]: When generated code is in the build folder:
1) It is clear that it is generated code
2) The code is not versioned in git and therefore the PRs are much much smaller.
3) When you auto-generate the code from a remote XSD file then anytime our logic is not in accordance with the latest XSD file we will get a compile error instead of a runtime error. Compile errors are much easier. to fix.
4) Having generated code injected in your source code, might lead to leakages (i.e. more code outside this module will start using the generated code) This will lead to major problems when the XSD will be updated.

File: create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusS3Client.java [hunk 16]
```diff
@@ -0,0 +1,98 @@
...
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import nva.commons.core.Environment;
+import nva.commons.core.JacocoGenerated;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ScopusS3Client {
+
+    private static final Logger logger = LoggerFactory.getLogger(ScopusS3Client.class);
```
[the remark this answers is not in the document]
Comment (gregorgabriel) [reply create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusS3Client.java:22]: Ok. I wasn't aware of that S3Driver.

File: create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusS3Client.java [hunk 17]
```diff
@@ -0,0 +1,98 @@
...
+            logger.error(CANNOT_CONNECT_TO_S3, e);
+        }
+    }
+
+    /**
+     * Get single file from s3 by filename.
+     * @param filename name of the file to be retrieved
+     * @return file as inputStream
+     */
+    protected InputStream getFile(String filename) {
```
[the remark this answers is not in the document]
Comment (gregorgabriel) [reply create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusS3Client.java:71]: What is a UnixPath?

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java (hunk 11 above)
Comment (gregorgabriel) [reply create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java:35]: Maybe it might never occur, but JaCoco 100% demands it to be tested. 
You want the Scenarios be testet at Handler-level. But you need to test your functionality (that single classes provide) where it is provided. There you can test the behavior of the particular class.

Besides is `handler.processInput(null, null, null)` just a placeholder, until we have implemented something. But teh tight JaCoco and PMD rules forces us to write something up front.

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java (hunk 12 above)
Comment (gregorgabriel) [reply create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java:67]: Thats copy-paste code from FetchDoiHandler. We will probably not even have Summary.class here.

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java (hunk 13 above)
Comment (gregorgabriel) [reply create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java:78]: Yes, placeholder code here as well.

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusS3ClientTest.java (hunk 14 above)
Comment (gregorgabriel) [reply create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusS3ClientTest.java:49]: In which package do you get access to that FakeS3Client? And why not just mcok it?

File: create-publication-from-scopus/build.gradle (hunk 3 above)
Comment (gregorgabriel) [reply create-publication-from-scopus/build.gradle:3]: Right. that is convincing.

File: gradle/libs.versions.toml (hunk 8 above)
Comment (gregorgabriel) [reply gradle/libs.versions.toml:6]: I don't think it is a good idea o use different versions of sdk in different modules. I suggest fetch-doi needs a major upgrade first.

Comment (gregorgabriel) [reply gradle/libs.versions.toml:6]: How should we handle issues like this? 
This is a scope-creep from out Sentralimport perspektive.
Is that a task on NVA board? Do you feel comfortable to outsource the upgrade to us?

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusS3ClientTest.java (hunk 7 above)
Comment (gregorgabriel) [reply create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusS3ClientTest.java:23]: Ok. We will follow your premisses for testing.

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java [hunk 18]
```diff
@@ -0,0 +1,121 @@
...
+     */
+    @BeforeEach
+    public void setUp() {
+        environment = mock(Environment.class);
+        context = getMockContext();
+        when(environment.readEnv(ApiGatewayHandler.ALLOWED_ORIGIN_ENV)).thenReturn(ALL_ORIGINS);
+    }
+
+    @Test
+    public void testResponseWithEmptyRequest()
```
[the remark this answers is not in the document]
Comment (gregorgabriel) [reply create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java:53]: We will follow your test premisses.

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java (hunk 9 above)
Comment (gregorgabriel) [reply create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java:62]: We will follow your test premisses.

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java (hunk 10 above)
Comment (gregorgabriel) [reply create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java:63]: We will follow your test premisses.

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java (hunk 12 above)
Comment (gregorgabriel) [reply create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java:67]: Should be fixed in FetchDoiHandler aswell.

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java (hunk 11 above)
Comment (gregorgabriel) [reply create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java:35]: We will follow your test premisses.

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/PublicationPersistenceServiceTest.java (hunk 6 above)
Comment (gregorgabriel) [reply create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/PublicationPersistenceServiceTest.java:9]: We will follow your test premisses.

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/PublicationConverterTest.java (hunk 5 above)
Comment (gregorgabriel) [reply create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/PublicationConverterTest.java:6]: We will follow your test premisses.

File: create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/PublicationConverter.java [hunk 19]
```diff
@@ -0,0 +1,4 @@
+package no.sikt.nva.scopus;
+
```
[the remark this answers is not in the document]
Comment (gregorgabriel) [reply create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/PublicationConverter.java:2]: Jeg forstår at du be oss om å bruke TDD utvikling så lenge vi jobber inn i NVA koden?

File: template.yaml [hunk 20]
```diff
@@ -283,4 +289,74 @@ Resources:
...
+          resource=boto3.resource('s3')
+          zip_object = resource.Object(bucket_name=ZIP_BUCKET, key=ZIP_FILE)
+          buffer = BytesIO(zip_object.get()["Body"].read())
+          z = zipfile.ZipFile(buffer)
+          for filename in z.namelist():
+            response = client.put_object(
+              Body=z.open(filename).read(),
+              Bucket=ZIP_BUCKET if filename.endswith(".zip") else XML_BUCKET,
+              Key=f'unzipped/{filename}' if filename.endswith(".zip") else filename
+            )
```
Comment (brinxmat) [inline template.yaml:321]: - Indent four spaces to comply with PEP8
- Use lowercase variable names (PEP8)
- What is happening with the "response" variable? Here, we just assign it multiple times without returning anything. The intention is the client.put_object, but we do nothing with the result.
  - Do we want to assess success across the processed files? (potentially retrying)
  - Do we want to report errors?

File: create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/PublicationConverter.java (hunk 19 above)
[the remark this answers is not in the document]
Comment (axthosarouris) [reply create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/PublicationConverter.java:2]: It is just a suggestion. From a reviewer's point of view I do not feel comfortable approving incomplete code that I do not know what it does. And I am not sure what this class is supposed to do and what is its purpose. I see that ScopusHandler depends on this class. 

So here I start to imagine things, probably they want to do this or that, so that is why they have added this class. But what the reviewer is imaging, is not what the coder imagined. So the reviewer approves code that has never existed and probably will never exist. 

In short, I cannot approve the existence of this class, since I do not know its purpose.

File: create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusHandler.java [hunk 21]
```diff
@@ -0,0 +1,95 @@
...
+        } catch (IOException e) {
+            throw new RuntimeException("Error creating handler", e);
+        }
+    }
+
+    @Override
+    protected Summary processInput(Void input, RequestInfo requestInfo, Context context) throws ApiGatewayException {
+        // Todo: listFiles kan droppes. Det gjør vi i et eget Lambda, som sender oss hvert filnavn som må prosesseres
+        // read S3 and get a list of all latest Files
+        List<String> filenames = s3Client.listFiles("latest?");
```
Comment (axthosarouris) [inline create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusHandler.java:69]: replace magic string with an explanatory constant

File: create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusS3Client.java [hunk 22]
```diff
@@ -0,0 +1,98 @@
...
+    public ScopusS3Client(AmazonS3 amazonS3Client, String bucketName) {
+        this.amazonS3Client = amazonS3Client;
+        this.bucketName = bucketName;
+    }
+
+    @JacocoGenerated
+    private void initS3Client(Environment environment) {
+        try {
+            this.amazonS3Client = AmazonS3ClientBuilder.standard()
+                    .withRegion(environment.readEnv(AWS_REGION))
```
Comment (axthosarouris) [inline create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusS3Client.java:57]: the environment variable can be read in directly when we you are creating the constant.

File: create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusS3Client.java (hunk 17 above)
[the remark this answers is not in the document]
Comment (axthosarouris) [reply create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusS3Client.java:71]: UnixPath is class for manipulating paths of the form `/some/folder/some/file` and it is supposed to work for both independent of the underlying OS. Basically it is like Java's Path but not affected by the OS

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusS3ClientTest.java (hunk 14 above)
Comment (axthosarouris) [reply create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusS3ClientTest.java:49]: Usually when we mock something, we use Mockito's any() methods. This means that your test will return the "expected" file even when the path you are requesting is wrong. FakeS3Client enables you to just store a file and then retrieve it as you would have done with the normal s3 client. But this is just a suggestion. The fake clients are in the nva-testutils package

File: gradle/libs.versions.toml (hunk 8 above)
Comment (axthosarouris) [reply gradle/libs.versions.toml:6]: use skd1

**Review by axthosarouris — CHANGES_REQUESTED** [review]

File: create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusS3Client.java (hunk 16 above)
[the remark this answers is not in the document]
Comment (gregorgabriel) [reply create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusS3Client.java:22]: Sorry, jeg finner ikke S3Driver noen plass i com.github.bibsysdev.
Kan du gir meg en hint, hvilke dependency jeg burde har. (På nett finnes det heller ingenting.)

[the remark this answers is not in the document]
Comment (gregorgabriel) [reply create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusS3Client.java:22]: Ok, da har jeg faktisk fjernet hele S3Client og bruker S3Driver direkte i Handler.
